v2.2/architecture/life-of-a-distributed-transaction.md

```diff
@@ -0,0 +1,204 @@
+
+This request is sent over a connection to your CockroachDB cluster that's established using a PostgreSQL driver.
+
+## Load Balancing & Routing
+
+Modern architectures require distributing your cluster across machines to improve throughput, latency, and uptime. This means queries are routed through load balancers, which choose the best CockroachDB node to connect to. Because all CockroachDB nodes have perfectly symmetrical access to data, this means your load balancer can connect your client to any node in the cluster and access any data while still guaranteeing strong consistency.
+
+Your architecture might also have additional layers of routing to enforce regulatory compliance, such as ensuring GDPR compliance.
+
+Once your router and load balancer determine the best node, your client's connection is established to the gateway node.
+
+## Gateway
+
+The gateway node handles the connection with the client, both receiving and responding to the request.
+
+### Parsing
+
+The gateway node first [parses](sql-layer.html#sql-parser-planner-executor) the client's SQL statement to ensure it's valid according to the CockroachDB dialect of PostgreSQL.
+
+While CockroachDB presents a SQL interface to clients, the actual database is built on top of a key-value store. Parsed SQL statements are converted into key-value operations, and a the node [generates a plan](sql-layer.html#logical-planning) to execute the query.
+
+### TxnCoordSender
+
+The newly generated plan is sent to the `TxnCoordSender`, which performs a large amount of the accounting and tracking for a transaction, including:
+
+- Accounts for all keys involved in a transaction. This is used, among other ways, to manage the transaction's state.
+- Packages all key-value operations into a `BatchRequest`, which are forwarded on to the node's `DistSender`.
+
+### DistSender
+
+The gateway node's `DistSender` receives `BatchRequests` from the `TxnCoordSender`. It dismantles the initial `BatchRequest` by taking each operation and finding which physical machine should receive the request for the range––known as the range's leaseholder. The address of the range's current leaseholder is readily available in both local caches, as well as in the [cluster's `meta` ranges](distribution-layer.html#meta-range-kv-structure).
+
+These dismantled `BatchRequests` are reassembled into new `BatchRequests` containing the address of the range's leaseholder.
+
+All writes operations also propagate the leaseholder's address back to the `TxnCoordSender`, so it can track and clean up write operations as necessary.
+
+The `DistSender` sends out the first `BatchRequest` for each range in parallel. As soon as it receives a provisional acknowledgment from the leasholder node’s evaluator (details below), it sends out the next `BatchRequest` for that range.
+
+The DistSender then waits to receive commit acknowledgments for all of its write operations, as well as values for all of its read operations.
```

Comment:
Consider removing "commit" from the term "commit acknowledgments". The reason for this is that commit means different things in different layers of CockroachDB. The DistSender might be performing an operation within an ongoing SQL transaction and will still wait for an acknowledgment.